### PR TITLE
Expose rate limiting errors from gpu enablement check

### PR DESCRIPTION
**Pull request by chrisdostert**
This PR passes through an error condition that was being ignored while testing if the machine is gpu enabled. Generally the only way this error occurs is during rate limiting when the image is not cached, which is a rather unlikely edge case, but are hitting it ourselves in e2e tests due to no not caching any images across test scenarios.

### CODE REVIEW

File: cli/.opspec/test/e2e/op.yml [hunk 1]
```diff
@@ -37,5 +41,7 @@ run:
                   /args.yml: $(scenario.call.scope)
                   /cmd.sh:
                   /usr/local/bin/opctl: $(../../../opctl-linux-amd64)
+                sockets:
+                  /var/run/docker.sock: $(dockerSocket)
```
Comment (chrisdostert) [inline cli/.opspec/test/e2e/op.yml:45]: this adds image caching (via reusing the same docker daemon across all tests)

File: sdks/go/node/containerruntime/docker/isGpuSupported.go [hunk 2]
```diff
@@ -54,6 +54,9 @@ func isGpuSupported(
 		"",
 		noOpEventPublisher{},
 	)
+	if err != nil {
```
Comment (chrisdostert) [inline sdks/go/node/containerruntime/docker/isGpuSupported.go:57]: this is the actual fix for the lack of rate limiting error surfacing during gpu testing

**Review by rajivbaviskar — APPROVED** [review]
